### ISSUE TITLE
[JENKINS-10944] MatrixProject can run on a heavyweight executor, leading to deadlocks

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-15#4749]
<p>Sometimes jobs get blocked by each other or "deadlock". We must manually cancel and restart the builds. </p>

<p>IRC Transcript<br/>
--------------<br/>
amrox: Hello. I had a brief exchange with @jenkins on twitter yesterday <a href="http://dl.dropbox.com/u/45634/Screen%20Shot%202011-09-08%20at%205.18.05%20PM.png" class="external-link" target="_blank" rel="nofollow noopener">http://dl.dropbox.com/u/45634/Screen%20Shot%202011-09-08%20at%205.18.05%20PM.png</a><br/>
amrox: my jenkins fell into a "deadlock" state again<br/>
amrox: it's still in that state now. what information can I provide?<br/>
rtyler: *nudges kohsuke *<br/>
rtyler: amrox: I'm @jenkinsci FWIW<br/>
amrox: rtyler: hi, and thanks for the responses yesterday<br/>
amrox: heres a screen recording... best way I could think to show the issue <a href="http://dl.dropbox.com/u/45634/Screen%20Recording%204.mov" class="external-link" target="_blank" rel="nofollow noopener">http://dl.dropbox.com/u/45634/Screen%20Recording%204.mov</a><br/>
farshidghods left the room (quit: Quit: Leaving.). (5:30:12 PM)<br/>
mconigliaro left the room (quit: Quit: mconigliaro). (5:30:16 PM)<br/>
amrox: I think it can be solved by just adding another executor<br/>
amrox: but I'd like to avoid that, and it seems like a bug?<br/>
kohsuke: amrox: we need thread dump. see <a href="https://wiki.jenkins-ci.org/display/JENKINS/Build+is+hanging" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins-ci.org/display/JENKINS/Build+is+hanging</a><br/>
amrox: jenkins master dump: <a href="http://dl.dropbox.com/u/45634/threaddump.html" class="external-link" target="_blank" rel="nofollow noopener">http://dl.dropbox.com/u/45634/threaddump.html</a><br/>
amrox: slave: <a href="http://dl.dropbox.com/u/45634/slave-threaddump.html" class="external-link" target="_blank" rel="nofollow noopener">http://dl.dropbox.com/u/45634/slave-threaddump.html</a><br/>
amrox: is that format acceptable? helpful at all?<br/>
amrox: should I just file a bug?<br/>
kennethreitz <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kennethre%40c-24-127-96-129.hsd1.va.comcast.net" class="user-hover" rel="kennethre@c-24-127-96-129.hsd1.va.comcast.net">kennethre@c-24-127-96-129.hsd1.va.comcast.net</a> entered the room. (5:45:36 PM)<br/>
kohsuke: amrox: yes, that'd be great<br/>
kohsuke: jenkins-admin: create ant-plugin on github for kohsuke<br/>
kohsuke: jenkins-admin: create javadoc-plugin on github for kohsuke<br/>
amrox: kohsuke: will do thanks<br/>
kohsuke: Is "content_viewer_ios_develop_build" the hanging job?<br/>
kohsuke: OK, so the issue is that the matrix parents are blocking the execution of its child builds<br/>
kohsuke: but the parent is also waiting for the completion of the child builds, hence the dead lock<br/>
kohsuke: amrox: ^^ did I get that right?<br/>
kohsuke: the question is why content_asset_verify_auto build is occupying an executor<br/>
amrox: yes that seems accurate<br/>
kohsuke: content_viewer_ios_develop_build is correctly using a temporary flyweight executor<br/>
kohsuke: ... as seen by the lack of number in the executor table<br/>
kohsuke: amrox: I assume all those builds are tried to remote-macslave-1<br/>
amrox: kohsuke: yes<br/>
kohsuke: OK. We'll capture this in the ticket you'll create<br/>
kohsuke: Thanks for bringing this to our attention, and sorry for the bug<br/>
amrox: thanks for building and maintaining jenkins ￼</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=amrox">amrox</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-10944" target="_blank">MatrixProject can run on a heavyweight executor, leading to deadlocks</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jglick.png" title="jglick's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jglick">jglick</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Major
<li><b>component(s)</b>: core
<li><b>label(s)</b>: 1.596.2-fixed, deadlock, flyweight, queue
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2015-01-09T13:00:10+00:00
<li><b>votes</b>: 1
<li><b>watchers</b>: 4
<li><b>imported</b>: 2025-11-16
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Sometimes jobs get blocked by each other or "deadlock". We must manually cancel and restart the builds. </p>

<p>IRC Transcript
--------------
amrox: Hello. I had a brief exchange with @jenkins on twitter yesterday <a href="http://dl.dropbox.com/u/45634/Screen%20Shot%202011-09-08%20at%205.18.05%20PM.png" class="external-link" target="_blank" rel="nofollow noopener">http://dl.dropbox.com/u/45634/Screen%20Shot%202011-09-08%20at%205.18.05%20PM.png</a>
amrox: my jenkins fell into a "deadlock" state again
amrox: it's still in that state now. what information can I provide?
rtyler: *nudges kohsuke *
rtyler: amrox: I'm @jenkinsci FWIW
amrox: rtyler: hi, and thanks for the responses yesterday
amrox: heres a screen recording... best way I could think to show the issue <a href="http://dl.dropbox.com/u/45634/Screen%20Recording%204.mov" class="external-link" target="_blank" rel="nofollow noopener">http://dl.dropbox.com/u/45634/Screen%20Recording%204.mov</a>
farshidghods left the room (quit: Quit: Leaving.). (5:30:12 PM)
mconigliaro left the room (quit: Quit: mconigliaro). (5:30:16 PM)
amrox: I think it can be solved by just adding another executor
amrox: but I'd like to avoid that, and it seems like a bug?
kohsuke: amrox: we need thread dump. see <a href="https://wiki.jenkins-ci.org/display/JENKINS/Build+is+hanging" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins-ci.org/display/JENKINS/Build+is+hanging</a>
amrox: jenkins master dump: <a href="http://dl.dropbox.com/u/45634/threaddump.html" class="external-link" target="_blank" rel="nofollow noopener">http://dl.dropbox.com/u/45634/threaddump.html</a>
amrox: slave: <a href="http://dl.dropbox.com/u/45634/slave-threaddump.html" class="external-link" target="_blank" rel="nofollow noopener">http://dl.dropbox.com/u/45634/slave-threaddump.html</a>
amrox: is that format acceptable? helpful at all?
amrox: should I just file a bug?
kennethreitz <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kennethre%40c-24-127-96-129.hsd1.va.comcast.net" class="user-hover" rel="kennethre@c-24-127-96-129.hsd1.va.comcast.net">kennethre@c-24-127-96-129.hsd1.va.comcast.net</a> entered the room. (5:45:36 PM)
kohsuke: amrox: yes, that'd be great
kohsuke: jenkins-admin: create ant-plugin on github for kohsuke
kohsuke: jenkins-admin: create javadoc-plugin on github for kohsuke
amrox: kohsuke: will do thanks
kohsuke: Is "content_viewer_ios_develop_build" the hanging job?
kohsuke: OK, so the issue is that the matrix parents are blocking the execution of its child builds
kohsuke: but the parent is also waiting for the completion of the child builds, hence the dead lock
kohsuke: amrox: ^^ did I get that right?
kohsuke: the question is why content_asset_verify_auto build is occupying an executor
amrox: yes that seems accurate
kohsuke: content_viewer_ios_develop_build is correctly using a temporary flyweight executor
kohsuke: ... as seen by the lack of number in the executor table
kohsuke: amrox: I assume all those builds are tried to remote-macslave-1
amrox: kohsuke: yes
kohsuke: OK. We'll capture this in the ticket you'll create
kohsuke: Thanks for bringing this to our attention, and sorry for the bug
amrox: thanks for building and maintaining jenkins ￼</p></pre>
</details>
</details>
<details><summary><i>environment</i></summary>

```
jenkins running on linux host<br/>
builds tied to remote mac host<br/>
have several matrix builds
```
</details>
<details><summary><i>2 attachments</i></summary>

- [slave-threaddump.html](https://issues.jenkins.io/secure/attachment/20821/slave-threaddump.html)
- [threaddump.html](https://issues.jenkins.io/secure/attachment/20820/threaddump.html)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>timwood0's <a href="https://issues.jenkins.io/browse/JENKINS-10944?focusedId=216539&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-216539">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>OK, three observations. <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>
</pre>
</details>

</details>
<p>OK, three observations. <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>timwood0's <a href="https://issues.jenkins.io/browse/JENKINS-10944?focusedId=216581&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-216581">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>With my current plugin updates (msg me for list), the problem has gone away, <b>except</b> in this use case:
1. Quiet Jenkins (announce "shut down")
2. Wait for all running builds to complete on all nodes
3. Queue a build of the matrix job
4. Cancel shutdown
5. Job begins building on all nodes except the one running the flyweight launcher task; build for that node remains queued in the deadlock condition.</p>

<p>So far it works to exchange steps 3 &amp; 4, and hope nothing else gets queued to run between them.</p>
</pre>
</details>

</details>
<p>With my current plugin updates (msg me for list), the problem has gone away, <b>except</b> in this use case:<br/>
1. Quiet Jenkins (announce "shut down")<br/>
2. Wait for all running builds to complete on all nodes<br/>
3. Queue a build of the matrix job<br/>
4. Cancel shutdown<br/>
5. Job begins building on all nodes except the one running the flyweight launcher task; build for that node remains queued in the deadlock condition.</p>

<p>So far it works to exchange steps 3 & 4, and hope nothing else gets queued to run between them.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck's <a href="https://issues.jenkins.io/browse/JENKINS-10944?focusedId=217122&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-217122">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jglick" class="user-hover" rel="jglick">jglick</a> Isn't this just a duplicate of <a href="https://issues.jenkins.io/browse/JENKINS-24519" title="Flyweight tasks only use one-off executor when they can be scheduled immediately" class="issue-link" data-issue-key="JENKINS-24519"><del>JENKINS-24519</del></a>? The other has more watchers and a better network of related/duplicate issues, so I'd rather resolve this.</p>

<p>(Also, ZD-21888.)</p>
</pre>
</details>

</details>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jglick" class="user-hover" rel="jglick">jglick</a> Isn't this just a duplicate of <a href="https://issues.jenkins.io/browse/JENKINS-24519" title="Flyweight tasks only use one-off executor when they can be scheduled immediately" class="issue-link" data-issue-key="JENKINS-24519"><del>JENKINS-24519</del></a>? The other has more watchers and a better network of related/duplicate issues, so I'd rather resolve this.</p>

<p>(Also, ZD-21888.)</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>timwood0's <a href="https://issues.jenkins.io/browse/JENKINS-10944?focusedId=217123&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-217123">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>It's not clear a fix to the other would fix this.  Can we make this a "see also," describing another use case, rather than "duplicate?"  Discussion on this general issue goes back years.</p>
</pre>
</details>

</details>
<p>It's not clear a fix to the other would fix this.  Can we make this a "see also," describing another use case, rather than "duplicate?"  Discussion on this general issue goes back years.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jglick.png" title="jglick's avatar" /> jglick's <a href="https://issues.jenkins.io/browse/JENKINS-10944?focusedId=217131&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-217131">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>If I understand <a href="https://issues.jenkins.io/browse/JENKINS-24519" title="Flyweight tasks only use one-off executor when they can be scheduled immediately" class="issue-link" data-issue-key="JENKINS-24519"><del>JENKINS-24519</del></a> correctly, that is a bug with similar symptoms, but a different cause, and potentially distinct fixes.</p>

<p>One change that would be helpful for diagnosing bugs of this category is to throw an <tt>IllegalStateException</tt> (or <tt>assert</tt>) if a <tt>FlyweightTask</tt> is ever scheduled onto a heavyweight executor, or vice-versa.</p>
</pre>
</details>

</details>
<p>If I understand <a href="https://issues.jenkins.io/browse/JENKINS-24519" title="Flyweight tasks only use one-off executor when they can be scheduled immediately" class="issue-link" data-issue-key="JENKINS-24519"><del>JENKINS-24519</del></a> correctly, that is a bug with similar symptoms, but a different cause, and potentially distinct fixes.</p>

<p>One change that would be helpful for diagnosing bugs of this category is to throw an <tt>IllegalStateException</tt> (or <tt>assert</tt>) if a <tt>FlyweightTask</tt> is ever scheduled onto a heavyweight executor, or vice-versa.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck's <a href="https://issues.jenkins.io/browse/JENKINS-10944?focusedId=217145&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-217145">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jglick" class="user-hover" rel="jglick">jglick</a>: I'm not so sure. It really looks like the issue in <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=timwood0" class="user-hover" rel="timwood0">timwood0</a>'s last comment is the same <a href="https://issues.jenkins.io/browse/JENKINS-24519" title="Flyweight tasks only use one-off executor when they can be scheduled immediately" class="issue-link" data-issue-key="JENKINS-24519"><del>JENKINS-24519</del></a>: When Jenkins is quieting down, FlyWeight tasks aren't scheduled, and use a real executor once they are.</p>

<p>The change you link to, and the one I link to in <a href="https://issues.jenkins.io/browse/JENKINS-24519" title="Flyweight tasks only use one-off executor when they can be scheduled immediately" class="issue-link" data-issue-key="JENKINS-24519"><del>JENKINS-24519</del></a> are essentially the same code.</p>

<p>Note that IMO it makes sense to not schedule Flyweight tasks while quieting down if they're not NonBlockingTask, however it makes no sense for them to be treated like normal ("heavyweight") tasks afterwards.</p>
</pre>
</details>

</details>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jglick" class="user-hover" rel="jglick">jglick</a>: I'm not so sure. It really looks like the issue in <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=timwood0" class="user-hover" rel="timwood0">timwood0</a>'s last comment is the same <a href="https://issues.jenkins.io/browse/JENKINS-24519" title="Flyweight tasks only use one-off executor when they can be scheduled immediately" class="issue-link" data-issue-key="JENKINS-24519"><del>JENKINS-24519</del></a>: When Jenkins is quieting down, FlyWeight tasks aren't scheduled, and use a real executor once they are.</p>

<p>The change you link to, and the one I link to in <a href="https://issues.jenkins.io/browse/JENKINS-24519" title="Flyweight tasks only use one-off executor when they can be scheduled immediately" class="issue-link" data-issue-key="JENKINS-24519"><del>JENKINS-24519</del></a> are essentially the same code.</p>

<p>Note that IMO it makes sense to not schedule Flyweight tasks while quieting down if they're not NonBlockingTask, however it makes no sense for them to be treated like normal ("heavyweight") tasks afterwards.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jglick.png" title="jglick's avatar" /> jglick's <a href="https://issues.jenkins.io/browse/JENKINS-10944?focusedId=217167&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-217167">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>The reproduction case for <a href="https://issues.jenkins.io/browse/JENKINS-24519" title="Flyweight tasks only use one-off executor when they can be scheduled immediately" class="issue-link" data-issue-key="JENKINS-24519"><del>JENKINS-24519</del></a> does not involve quieting down, it involves a slave being temporarily offline. But you may be right that the faulty code path and thus fix is the same: do not proceed to <tt>p.enter(this)</tt> when <tt>p.task instanceof FlyweightTask</tt>. I am not sure what should be done instead: from the callers of <tt>makeBuildable</tt> it seems that merely returning without doing anything is not correct either, that we would need to actually revert to the waiting state. Will have to examine this a bit more closely, starting by reproducing at least one of the failure modes in a test.</p>

<blockquote><p>it makes sense to not schedule Flyweight tasks while quieting down if they're not NonBlockingTask, however it makes no sense for them to be treated like normal ("heavyweight") tasks afterwards</p></blockquote>

<p>Agreed on both points.</p>
</pre>
</details>

</details>
<p>The reproduction case for <a href="https://issues.jenkins.io/browse/JENKINS-24519" title="Flyweight tasks only use one-off executor when they can be scheduled immediately" class="issue-link" data-issue-key="JENKINS-24519"><del>JENKINS-24519</del></a> does not involve quieting down, it involves a slave being temporarily offline. But you may be right that the faulty code path and thus fix is the same: do not proceed to <tt>p.enter(this)</tt> when <tt>p.task instanceof FlyweightTask</tt>. I am not sure what should be done instead: from the callers of <tt>makeBuildable</tt> it seems that merely returning without doing anything is not correct either, that we would need to actually revert to the waiting state. Will have to examine this a bit more closely, starting by reproducing at least one of the failure modes in a test.</p>

<blockquote><p>it makes sense to not schedule Flyweight tasks while quieting down if they're not NonBlockingTask, however it makes no sense for them to be treated like normal ("heavyweight") tasks afterwards</p></blockquote>

<p>Agreed on both points.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck's <a href="https://issues.jenkins.io/browse/JENKINS-10944?focusedId=217188&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-217188">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Jesse: You're right. I've accessed that issue so often and stopped reading the description a long time ago when I clearly shouldn't have, instead interpreting it from the code I've linked to and the behavior most relevant for me. Sorry about that.</p>

<p>It seems to me that <tt>Queue.Item.task</tt>'s flyweight property needs to be relevant somewhere else in addition to when it enters the queue (possibly <tt>Queue.maintain()</tt> around <a href="https://github.com/jenkinsci/jenkins/blob/7156e4470f05a9c41015be78d827dd1be0c02fca/core/src/main/java/hudson/model/Queue.java#L1046" class="external-link" target="_blank" rel="nofollow noopener">here</a>?) as well.</p>
</pre>
</details>

</details>
<p>Jesse: You're right. I've accessed that issue so often and stopped reading the description a long time ago when I clearly shouldn't have, instead interpreting it from the code I've linked to and the behavior most relevant for me. Sorry about that.</p>

<p>It seems to me that <tt>Queue.Item.task</tt>'s flyweight property needs to be relevant somewhere else in addition to when it enters the queue (possibly <tt>Queue.maintain()</tt> around <a href="https://github.com/jenkinsci/jenkins/blob/7156e4470f05a9c41015be78d827dd1be0c02fca/core/src/main/java/hudson/model/Queue.java#L1046" class="external-link" target="_blank" rel="nofollow noopener">here</a>?) as well.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/dogfood.png" title="dogfood's avatar" /> dogfood's <a href="https://issues.jenkins.io/browse/JENKINS-10944?focusedId=219072&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-219072">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Integrated in <span class="image-wrap" style=""><img src="http://ci.jenkins-ci.org/images/16x16/blue.png" style="border: 0px solid black" /></span> <a href="http://ci.jenkins-ci.org/job/jenkins_main_trunk/3924/" class="external-link" target="_blank" rel="nofollow noopener">jenkins_main_trunk #3924</a>
     <span class="error">&#91;FIXED JENKINS-10944&#93;</span> <span class="error">&#91;FIXED JENKINS-24519&#93;</span> If makeBuildable fails on a FlyweightTask, keep it in queue. (Revision 3e344a94a9eed316d0c351becb08287b473b6521)</p>

<p>     Result = SUCCESS
jesse glick : <a href="https://github.com/jenkinsci/jenkins/commit/3e344a94a9eed316d0c351becb08287b473b6521" class="external-link" target="_blank" rel="nofollow noopener">3e344a94a9eed316d0c351becb08287b473b6521</a>
Files : </p>
<ul>
	<li>test/src/test/java/hudson/model/QueueTest.java</li>
	<li>core/src/main/java/hudson/model/Queue.java</li>
</ul>

</pre>
</details>

</details>
<p>Integrated in <span class="image-wrap" style=""><img src="http://ci.jenkins-ci.org/images/16x16/blue.png" style="border: 0px solid black" /></span> <a href="http://ci.jenkins-ci.org/job/jenkins_main_trunk/3924/" class="external-link" target="_blank" rel="nofollow noopener">jenkins_main_trunk #3924</a><br/>
     <span class="error">&#91;FIXED JENKINS-10944&#93;</span> <span class="error">&#91;FIXED JENKINS-24519&#93;</span> If makeBuildable fails on a FlyweightTask, keep it in queue. (Revision 3e344a94a9eed316d0c351becb08287b473b6521)</p>

<p>     Result = SUCCESS<br/>
jesse glick : <a href="https://github.com/jenkinsci/jenkins/commit/3e344a94a9eed316d0c351becb08287b473b6521" class="external-link" target="_blank" rel="nofollow noopener">3e344a94a9eed316d0c351becb08287b473b6521</a><br/>
Files : </p>
<ul>
	<li>test/src/test/java/hudson/model/QueueTest.java</li>
	<li>core/src/main/java/hudson/model/Queue.java</li>
</ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/dogfood.png" title="dogfood's avatar" /> dogfood's <a href="https://issues.jenkins.io/browse/JENKINS-10944?focusedId=237650&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-237650">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Integrated in <span class="image-wrap" style=""><img src="http://ci.jenkins-ci.org/images/16x16/yellow.png" style="border: 0px solid black" /></span> <a href="http://ci.jenkins-ci.org/job/jenkins_main_trunk/4292/" class="external-link" target="_blank" rel="nofollow noopener">jenkins_main_trunk #4292</a>
     <a href="https://issues.jenkins.io/browse/JENKINS-10944" title="MatrixProject can run on a heavyweight executor, leading to deadlocks" class="issue-link" data-issue-key="JENKINS-10944"><del>JENKINS-10944</del></a> Merge branch 'master' into FlyweightTask-<a href="https://issues.jenkins.io/browse/JENKINS-10944" title="MatrixProject can run on a heavyweight executor, leading to deadlocks" class="issue-link" data-issue-key="JENKINS-10944"><del>JENKINS-10944</del></a> (Revision ff26bb8a18c62a4cd008cfff935166e38ee94f6a)</p>

<p>     Result = UNSTABLE
ogondza : <a href="https://github.com/jenkinsci/jenkins/commit/ff26bb8a18c62a4cd008cfff935166e38ee94f6a" class="external-link" target="_blank" rel="nofollow noopener">ff26bb8a18c62a4cd008cfff935166e38ee94f6a</a>
Files : </p>
<ul>
	<li>test/src/test/java/hudson/model/QueueTest.java</li>
	<li>core/src/main/java/hudson/model/Queue.java</li>
	<li>core/src/main/java/jenkins/model/Jenkins.java</li>
	<li>core/src/main/java/hudson/model/LoadBalancer.java</li>
</ul>

</pre>
</details>

</details>
<p>Integrated in <span class="image-wrap" style=""><img src="http://ci.jenkins-ci.org/images/16x16/yellow.png" style="border: 0px solid black" /></span> <a href="http://ci.jenkins-ci.org/job/jenkins_main_trunk/4292/" class="external-link" target="_blank" rel="nofollow noopener">jenkins_main_trunk #4292</a><br/>
     <a href="https://issues.jenkins.io/browse/JENKINS-10944" title="MatrixProject can run on a heavyweight executor, leading to deadlocks" class="issue-link" data-issue-key="JENKINS-10944"><del>JENKINS-10944</del></a> Merge branch 'master' into FlyweightTask-<a href="https://issues.jenkins.io/browse/JENKINS-10944" title="MatrixProject can run on a heavyweight executor, leading to deadlocks" class="issue-link" data-issue-key="JENKINS-10944"><del>JENKINS-10944</del></a> (Revision ff26bb8a18c62a4cd008cfff935166e38ee94f6a)</p>

<p>     Result = UNSTABLE<br/>
ogondza : <a href="https://github.com/jenkinsci/jenkins/commit/ff26bb8a18c62a4cd008cfff935166e38ee94f6a" class="external-link" target="_blank" rel="nofollow noopener">ff26bb8a18c62a4cd008cfff935166e38ee94f6a</a><br/>
Files : </p>
<ul>
	<li>test/src/test/java/hudson/model/QueueTest.java</li>
	<li>core/src/main/java/hudson/model/Queue.java</li>
	<li>core/src/main/java/jenkins/model/Jenkins.java</li>
	<li>core/src/main/java/hudson/model/LoadBalancer.java</li>
</ul>
